skbio/tree/_tree.py

```diff
@@ -617,6 +617,77 @@ def lowest_common_ancestor(self, tipnames):
 
     lca = lowest_common_ancestor  # for convenience
 
+    def path(self, other, include_ends=False):
+        r"""Return the list of nodes in the path from one node to another.
+
+        Parameters
+        ----------
+        other : TreeNode
+            Final node of path.
+        include_ends: bool, optional
+            Whether to include the initial and final nodes in the list.
+
+        Returns
+        -------
+        list
+            List of TreeNode objects.
+
+        Examples
+        --------
+        >>> from skbio import TreeNode
+        >>> tree = TreeNode.read(["((a,b)c,(d,e)f)root;"])
+        >>> node_1, node_2 = tree.find('a'), tree.find('d')
+        >>> path = node_1.path(node_2)
+        >>> print(len(path))
+        3
+        >>> path_2 = node_1.path(node_2, include_ends=True)
+        >>> print(len(path_2))
+        5
+
+        """
+        # check to see if nodes are on same tree
+        root1, root2 = self.root(), other.root()
+        if root1 is not root2:
+            raise TypeError("Could not find path between nodes.")
+
+        # create list of ancestors for both nodes
+        anc1, anc2 = (
+            [node for node in self.ancestors()],
+            [node for node in other.ancestors()],
+        )
+
+        # find lowest common ancestor
+        for n1, n2 in zip(reversed(anc1), reversed(anc2)):
+            if n1 is n2:
+                lca = n1
+            else:
+                break
+
+        # delete nodes from intersection of ancestors
+        del anc1[anc1.index(lca) :]
+        del anc2[anc2.index(lca) :]
+
+        # reverse the second list for correct order
+        anc2.reverse()
+
+        # handle cases for nodes that are ancestors of each other
+        if self in anc2:
+            del anc2[anc2.index(self)]
+        elif other in anc1:
+            del anc1[anc1.index(other)]
+        else:
+            anc1 = anc1 + [lca]
+
+        # create list of nodes in path
+        path_list = anc1 + anc2
+
+        # insert initial and final nodes if desired
+        if include_ends is True:
```

Comment:
It should be cheaper to slice a list than to insert an element to the beginning of a list. For example:

```python
if include_ends is False:
    path_list = path_list[1:-1]
```

Reply:
A Python list is a [stack](https://www.geeksforgeeks.org/difference-between-stack-and-queue-data-structures/). Adding / deleting elements to the end is efficient, but that to the beginning is inefficient.

Reply:
On a 2nd thought -- if `path_list` doesn't already contain head and tail, you may disregard my suggestion above.